src/test/integration/org/apache/cassandra/testing/CassandraTestTemplate.java

```diff
@@ -148,34 +149,21 @@ private BeforeEachCallback beforeEach()
                 int nodesPerDc = annotation.nodesPerDc();
                 int dcCount = annotation.numDcs();
                 int newNodesPerDc = annotation.newNodesPerDc(); // if the test wants to add more nodes later
+                Preconditions.checkArgument(newNodesPerDc >= 0,
+                                            "newNodesPerDc cannot be a negative number");
+                int originalNodeCount = nodesPerDc * dcCount;
                 int finalNodeCount = dcCount * (nodesPerDc + newNodesPerDc);
                 Versions.Version requestedVersion = versions.getLatest(new Semver(version.version(),
                                                                                   Semver.SemverType.LOOSE));
                 SimpleCassandraVersion versionParsed = SimpleCassandraVersion.create(version.version());
 
                 UpgradeableCluster.Builder clusterBuilder =
-                    UpgradeableCluster.build(nodesPerDc * dcCount)
-                                      .withVersion(requestedVersion)
-                                      .withDCs(dcCount)
-                                      .withDataDirCount(annotation.numDataDirsPerInstance())
-                                      .withConfig(config -> {
-                                      if (annotation.nativeTransport())
-                                      {
-                                          config.with(Feature.NATIVE_PROTOCOL);
-                                      }
-                                      if (annotation.jmx())
-                                      {
-                                          config.with(Feature.JMX);
-                                      }
-                                      if (annotation.gossip())
-                                      {
-                                          config.with(Feature.GOSSIP);
-                                      }
-                                      if (annotation.network())
-                                      {
-                                          config.with(Feature.NETWORK);
-                                      }
-                                  });
+                UpgradeableCluster.build(originalNodeCount)
+//                                  .withDynamicPortAllocation(true) // to allow parallel test runs
```

Comment:
Temporarily disabled parallel runs as it was not playing well with in-jvm dtest based integration tests

Reply:
we should be able to leverage this feature now. https://issues.apache.org/jira/browse/CASSANDRA-18722 has been merged